static/app/views/explore/hooks/useTraceItemDetails.tsx

```diff
@@ -0,0 +1,191 @@
+import {
+  type ApiQueryKey,
+  fetchDataQuery,
+  useApiQuery,
+  useQueryClient,
+} from 'sentry/utils/queryClient';
+import useOrganization from 'sentry/utils/useOrganization';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import useProjectFromId from 'sentry/utils/useProjectFromId';
+import {
+  getRetryDelay,
+  shouldRetryHandler,
+} from 'sentry/views/insights/common/utils/retryHandlers';
+
+const DEFAULT_REFERRER = 'use-trace-item-details';
+const DEFAULT_HOVER_TIMEOUT = 300;
+
+/**
+ * ProjectTraceItemDetailsEndpoint currently only supports ourlogs dataset
+ * TODO: Add SPANS_EAP once the backend supports it.
+ */
+type EAPDataset = DiscoverDatasets.OURLOGS;
+
+export interface UseTraceItemDetailsProps {
+  /**
+   * Trace items are only supported by EAP.
+   */
+  dataset: EAPDataset;
+  /**
+   * Every trace item belongs to a project.
+   */
+  projectId: string;
+  /**
+   * The trace item ID representing an EAP trace item.
+   */
+  traceItemId: string;
+  /**
+   * Alias for `enabled` in react-query.
+   */
+  enabled?: boolean;
+  /**
+   * Sets referrer parameter in the API Payload. Set of allowed referrers are defined
+   * as ALLOWED_EVENTS_REFERRERS on the backend.
+   */
+  referrer?: string;
+}
+
+export type TraceItemAttributes = Record<string, TraceItemResponseAttribute>;
+
+interface TraceItemDetailsResponse {
+  attributes: TraceItemAttributes;
+  itemId: string;
+  timestamp: string;
+}
+
+type TraceItemDetailsUrlParams = {
+  organizationSlug: string;
+  projectSlug: string;
+  traceItemId: string;
+};
+
+type TraceItemDetailsQueryParams = Pick<DiscoverQueryRequestParams, 'referrer'> & {
+  dataset: EAPDataset;
+};
+
+export type TraceItemResponseAttribute =
+  | {type: 'str'; value: string}
+  | {type: 'int'; value: number}
+  | {type: 'float'; value: number}
+  | {type: 'bool'; value: boolean};
+
+/**
+ * Query hook fetching trace item details in EAP.
+ */
+export function useTraceItemDetails(props: UseTraceItemDetailsProps) {
+  const organization = useOrganization();
+  const project = useProjectFromId({project_id: props.projectId});
+  const {isReady: pageFiltersReady} = usePageFilters();
+  const enabled = pageFiltersReady && (props.enabled ?? true) && !!project;
+
+  if (!project) {
+    captureException(
+      new Error(`Project "${props.projectId}" not found in useTraceItemDetails`)
+    );
+  }
+
+  const queryParams: TraceItemDetailsQueryParams = {
+    referrer: props.referrer,
+    dataset: props.dataset,
+  };
+
+  const queryOptions = {
+    enabled: enabled && pageFiltersReady,
+    retry: shouldRetryHandler,
+    retryDelay: getRetryDelay,
+    staleTime: Infinity,
+  };
+
+  const result = useApiQuery<TraceItemDetailsResponse>(
+    traceItemDetailsQueryKey({
+      urlParams: {
+        organizationSlug: organization.slug,
+        projectSlug: project?.slug ?? '',
+        traceItemId: props.traceItemId,
+      },
+      queryParams,
+    }),
+    queryOptions
```

Comment:
@TkDodo these should hopefully fix the rq eslint troubles? 😄

Reply:
yes, this looks good! One thing I think would be better would be to inline the `queryOptions` instead of extracting them to an extra const.

TypeScript doesn’t do excess property checking on separately defined objects, so if you mistype one of the options or react-query renames one of them (which happens from time to time), it won’t show up as an error here because the fields are all optional.

When objects are inlined, typescript does excess property checking, so we’d get an error for free. Alternatively, we could add `satisfies UseApiQueryOptions<TraceItemDetailsResponse>` to the `const queryOptions`, but I think inlining is better.

Reply:
FYI, I just merged the lint rule PR. sorry for the conflicts 😅